packages/devextreme/testing/testcafe/update_failed_etalons.mjs

```diff
@@ -0,0 +1,113 @@
+import {
+  readdirSync, lstatSync, existsSync, copyFileSync,
+} from 'fs';
+import { join, resolve } from 'path';
+import { argv } from 'process';
+
+const PATH_TO_CURRENT_ETALONS = argv[2];
+const PATH_TO_TESTCAFE_TESTS = './testing/testcafe/tests';
+const NEW_ETALON_FILE_NAME_PATTERN = /^(?!.*(?:_mask|_diff|_etalon)\.png$).*\.png$/i;
+const ETALONS_FOLDER_NAME = 'etalons';
+
+const getAllEtalonsNames = () => {
+  const newEtalons = new Map();
+  const allFileNames = readdirSync(PATH_TO_CURRENT_ETALONS);
+
+  allFileNames.forEach((name) => {
+    if (NEW_ETALON_FILE_NAME_PATTERN.test(name)) {
+      newEtalons.set(name, true);
+    }
+  });
+
+  return newEtalons;
+};
+
+const isDirectory = (fileName) => lstatSync(fileName).isDirectory();
+
+function getAllDirs(path) {
+  return readdirSync(path)
+    .map((fileName) => join(path, fileName))
+    .filter(isDirectory);
+}
+
+function processEtalonFolder(etalonFolderPath, allEtalons) {
+  const allFiles = readdirSync(etalonFolderPath);
+  let hasEtalonsToMove = allEtalons.size !== 0;
+
+  allFiles.forEach((currentFileName) => {
+    if (hasEtalonsToMove) {
```

Comment:
If we use `for` here we can just return when there is no rest etalons.
```javascript
if (allEtalons.size === 0) [
    return false;
}    
````
I see following advantages of this approach:
- We will exactly stop loop keeping the time
- It'll allow us to get rid of `hasEtalonsToMove` variable
- We'll reduce nesting
- It's clearer for me to see a concrete escaping condition because u don't need to check where and why variable is updated (it's especially useful for simplifying of the `processFolder` function)

Reply:
fixed